magento/pwa-studio
# [Bugfix] [Refactor] Order confirmation page

## Description

Fixes a bug with the order confirmation page where data would not persist between a page refresh.

NOTE: This has been fixed for authenticated users, but not for guest users.

This PR also refactors the order confirmation page component to be less coupled with the checkout component.

## Related Issue







Closes PWA-2484

## Acceptance





Any core developer

### Verification Stakeholders



Any core developer

### Specification




## Verification Steps





1. Go through checkout as a logged in user
2. On the order confirmation page, refresh the page
3. Verify order data is still rendered

#### Test scenario(s) for direct fix/feature





1. Verify guest users can still go through checkout steps
2. Verify authenticated users can go through checkout steps

#### Test scenario(s) for any existing impacted features/areas





#### Test scenario(s) for any Magento Backend Supported Configurations





#### Is Browser/Device testing needed?





#### Any ad-hoc/edge case scenarios that need to be considered?




Authenticated users logs out and guest user tries to navigate back to the order confirmation page.

## Screenshots / Screen Captures (if appropriate)

## Breaking Changes (if any)





## Checklist



-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.
